Cannot build project using chartjs if TypeScript compiler option `isolatedModules`  is set to `true`

### Expected behavior

In a project embedding `chart.js`, the TypeScript transpilation with the following options works successfully:
- `"isolatedModules": true`,
- `"skipLibCheck": false`.

### Current behavior

Transpilation fails because of en exported `const enum`:

```$ tsc
node_modules/chart.js/dist/types/index.d.ts:579:39 - error TS2748: Cannot access ambient const enums when the '--isolatedModules' flag is provided.

579 export type UpdateMode = keyof typeof UpdateModeEnum;
                                          ~~~~~~~~~~~~~~


Found 1 error in node_modules/chart.js/dist/types/index.d.ts:579
```

More explanation here: https://ncjamieson.com/dont-export-const-enums/

### Reproducible sample

https://stackblitz.com/edit/vitejs-vite-y2wwyb?file=tsconfig.json

### Optional extra steps/info to reproduce

Take a look at the terminal at the bottom of the page to see the transpilation error

### Possible solution

Make `UpdateModeEnum` a simple `enum`, not a `const` one maybe.

### Context

_No response_

### chart.js version

v4.2.0

### Browser name and version

_No response_

### Link to your project

_No response_

--- a/src/types/index.d.ts
+++ b/src/types/index.d.ts
@@ -566,17 +566,7 @@ export declare type ChartItem =
   | { canvas: HTMLCanvasElement }
   | ArrayLike<CanvasRenderingContext2D | HTMLCanvasElement>;
 
-export declare const enum UpdateModeEnum {
-  resize = 'resize',
-  reset = 'reset',
-  none = 'none',
-  hide = 'hide',
-  show = 'show',
-  default = 'default',
-  active = 'active'
-}
-
-export type UpdateMode = keyof typeof UpdateModeEnum;
+export type UpdateMode = 'resize' | 'reset' | 'none' | 'hide' | 'show' | 'default' | 'active';
 
 export declare class DatasetController<
   TType extends ChartType = ChartType,

--- a/src/types/index.d.ts
+++ b/src/types/index.d.ts
@@ -566,7 +566,17 @@ export declare type ChartItem =
   | { canvas: HTMLCanvasElement }
   | ArrayLike<CanvasRenderingContext2D | HTMLCanvasElement>;
 
-export type UpdateMode = 'resize' | 'reset' | 'none' | 'hide' | 'show' | 'default' | 'active';
+export declare enum UpdateModeEnum {
+  resize = 'resize',
+  reset = 'reset',
+  none = 'none',
+  hide = 'hide',
+  show = 'show',
+  default = 'default',
+  active = 'active'
+}
+
+export type UpdateMode = keyof typeof UpdateModeEnum;
 
 export declare class DatasetController<
   TType extends ChartType = ChartType,

